src/components/bookmark/BookmarkButton.tsx

```diff
@@ -19,6 +20,7 @@ import {
   BookmarkEvent,
   EVENT_BOOKMARK,
 } from "../map/mapbox/controls/menu/Definition";
+import { checkExtent } from "../../utils/GeoJsonUtils";
```

Comment:
No good to still keep the query , is this the function that create the collection?

    const getCollectionData = useCallback(
      async (uuid: string) => {
        return dispatch(fetchResultByUuidNoStore(uuid))
          .unwrap()
          .then((collection: OGCCollection) => collection)
          .catch((error: any) => {
            console.error("Error fetching collection data:", error);
            // TODO: handle error in ErrorBoundary
          });
      },
      [dispatch]
    );